app/controllers/admin_controller.rb

```diff
@@ -76,13 +79,11 @@ def check_is_admin
   # This will be used in other controllers in the admin section to be sure that
   # the user has the granualr permissions and is only used if the user is not an admin.
   def get_user_permission
-    if session[:cas_user]
-      user_permissions = UserPermission.find_by(username: session[:cas_user])
-      # set @user_libs and @user_depts if they are not an admin
-      if !check_is_admin && user_permissions != nil
-        @user_libs = clean_array user_permissions.libraries
-        @user_depts = clean_array user_permissions.departments
-      end
-    end
+    return unless session[:cas_user]
+    user_permissions = UserPermission.find_by(username: session[:cas_user])
+    # set @user_libs and @user_depts if they are not an admin
+    return unless !check_is_admin && !user_permissions.nil?
```

Comment:
What are you return here?  to me it looks like your returning nil if is an admin and if they have permissions